src/rgw/rgw_civetweb.cc

```diff
@@ -79,6 +79,12 @@ void RGWCivetWeb::init_env(CephContext *cct)
   for (int i = 0; i < info->num_headers; i++) {
     const struct mg_request_info::mg_header* header = &info->http_headers[i];
     const boost::string_ref name(header->name);
+
+    if (!header->value) {
```

Comment:
This is fine as-written, but also consider an explicit comparison to nullptr to leave no doubt as to what's happening here.